src/Controller/Component/CsrfComponent.php

```diff
@@ -61,6 +61,19 @@ class CsrfComponent extends Component
         'field' => '_csrfToken',
     ];
 
+    /**
+     * Warn if CsrfComponent is used together with CsrfProtectionMiddleware
+     *
+     * @param array $config The config data.
+     * @return void
+     */
+    public function initialize(array $config)
+    {
+        if ($this->getController()->request->getParam('_csrfToken') !== false) {
```

Comment:
Shouldn't this be using `getRequest()`?

Reply:
I've seen `@deprecated .... Use Component::getController() ... instead.` at `Component::request`, but forgot about `Controller::request` 😂

Anyway, this component is removed from `4.x` and wouldn't be a problem.